skrub/_table_vectorizer.py

```diff
@@ -921,9 +742,44 @@ def _check_X(self, X, reset):
                 " a minimum of 1 is required."
             )
         self._check_n_features(X, reset=reset)
+        # TODO: _check_feature_names raises a warning when fitting on dataframe
+        # but transforming on a numpy array.
+        # In practice, this looks error-prone and we need to discuss
+        # whether to raise an error instead.
+        #
+        # Note that when fitting on a dataframe and transforming on
+        # the same dataframe with different column names,
+        # _check_feature_names will raise an error.
+        self._check_feature_names(X, reset=reset)
+        feature_names = _get_feature_names(X)
+        feature_names_in = getattr(self, "feature_names_in_", None)
+        if feature_names is None and feature_names_in is not None:
+            X.columns = feature_names_in
 
         return X
 
+    def fit(self, X: ArrayLike, y: ArrayLike = None) -> "TableVectorizer":
+        """Fit all transformers using X.
+
+        Parameters
+        ----------
+        X : {array-like, dataframe} of shape (n_samples, n_features)
+            Input data, of which specified subsets are used to fit the
+            transformers.
+
+        y : array-like of shape (n_samples, ...), default=None
+            Targets for supervised learning.
+
+        Returns
+        -------
+        self : TableVectorizer
+            This estimator.
+        """
+        # we use fit_transform to make sure to set sparse_output_ (for which we
+        # need the transformed data) to have consistent output type in predict
+        self.fit_transform(X, y=y)
```

Comment:
Now that I look more closely at it, is it necessary?
We could rename our `fit_transform` to `fit` and either use `column_transformer.fit()` or `column_transformer.fit_transform()` since they are essentially the same thing. This would bring more clarity to the user (and also to us). WDYT?

Reply:
But the current `TableVectorizer.fit_transform` does not only `fit` (it `transform` as well). So I don't see why to rename it `fit`?

> either use column_transformer.fit() or column_transformer.fit_transform()

I am also a bit confused here because our `fit_transform` will validate some fitted attributes and we cannot solely rely on `column_transformer`, isn't it?

Reply:
Nevermind, works for me